### PR TITLE
Add support for joining as sleepy-end-device

### PR DESCRIPTION
**Pull request by abtink**
This commit adds support for joining as a sleepy-end-device. It
also adds support for getting/setting the data poll period through
the wpan property "NCP:SleepyPollInterval".

### CODE REVIEW

**Comment by abtink** [comment]
@darconeous can you please review this PR? Thanks.

**Comment by abtink** [comment]
Rebased on master

File: third_party/openthread/src/ncp/spinel.h [hunk 1]
```diff
@@ -636,16 +636,17 @@ typedef enum
...
-    SPINEL_PROP_MAC_SCAN_MASK           = SPINEL_PROP_MAC__BEGIN + 1, ///< [A(C)]
-    SPINEL_PROP_MAC_SCAN_PERIOD         = SPINEL_PROP_MAC__BEGIN + 2, ///< ms-per-channel [S]
-    SPINEL_PROP_MAC_SCAN_BEACON         = SPINEL_PROP_MAC__BEGIN + 3, ///< chan,rssi,mac_data,net_data [CcdD]
-    SPINEL_PROP_MAC_15_4_LADDR          = SPINEL_PROP_MAC__BEGIN + 4, ///< [E]
-    SPINEL_PROP_MAC_15_4_SADDR          = SPINEL_PROP_MAC__BEGIN + 5, ///< [S]
-    SPINEL_PROP_MAC_15_4_PANID          = SPINEL_PROP_MAC__BEGIN + 6, ///< [S]
-    SPINEL_PROP_MAC_RAW_STREAM_ENABLED  = SPINEL_PROP_MAC__BEGIN + 7, ///< [C]
-    SPINEL_PROP_MAC_PROMISCUOUS_MODE    = SPINEL_PROP_MAC__BEGIN + 8, ///< [C]
-    SPINEL_PROP_MAC_ENERGY_SCAN_RESULT  = SPINEL_PROP_MAC__BEGIN + 9, ///< chan,maxRssi [Cc]
+    SPINEL_PROP_MAC_SCAN_STATE          = SPINEL_PROP_MAC__BEGIN + 0,  ///< [C]
```
Comment (darconeous) [inline third_party/openthread/src/ncp/spinel.h]: Whitespace changes?

Comment (abtink) [inline third_party/openthread/src/ncp/spinel.h]: Yes. Aligning the start of comment.

**Pull request merged**